libtenzir/include/tenzir/detail/string.hpp

```diff
@@ -148,8 +146,24 @@ replace_all(std::string str, std::string_view search, std::string_view replace);
 /// of the string pointed to by `str`.
 /// @returns A vector of substrings.
 std::vector<std::string_view>
-split(std::string_view str, std::string_view sep, std::string_view esc = "",
-      size_t max_splits = -1, bool include_sep = false);
+split(std::string_view str, std::string_view sep, size_t max_splits = -1,
+      bool include_sep = false);
+
+/// Splits a character sequence into a vector of substrings, with escaping of
+/// the separator.
+/// @param str The string to split.
+/// @param sep The separator where to split.
+/// @param esc The escape string. If *esc* occurs immediately before
+///            *sep*, then *sep* will not count as a separator.
+///            In that case, *esc* will not be included in the output.
+/// @param max_splits The maximum number of splits to perform.
+/// @param include_sep If `true`, also include the separator after each
+///                    match, without including a possible escape string.
+/// @pre `!sep.empty() && !esc.empty()`
+/// @returns A vector of substrings, with the separator escape strings removed.
+std::vector<std::string>
+split_escaped(std::string_view str, std::string_view sep, std::string_view esc,
+              size_t max_splits = -1, bool include_sep = false);
```

Comment:
You're right, we seem to not be using it anywhere. I'll just axe it out, since I also think it's a little weird how it returns the separators as elements of their own.